addons/website_form/static/src/snippets/s_website_form/000.js

```diff
@@ -193,12 +193,16 @@ odoo.define('website_form.s_website_form', function (require) {
                     }
                     switch (successMode) {
                         case 'redirect':
-                            successPage = successPage.startsWith("/#") ? successPage.slice(1) : successPage;
+                            successPage = window.location.pathname === "/" && successPage.startsWith("/#") ? successPage.slice(1) : successPage;
```

Comment:
I am not too sure to understand what is trying to be achieved here (even before your diff).
Is the goal to detect that the success URL is the homepage and that we are on the homepage, and in this case don't perform a redirect but a scrollTo (note that the browser don't actually reload but do an ugly scrollto himself)?

If that's the case, why is this code not doing that for every page but only the homepage? :thinking: 
Very louch to do that only for homepage while it could work on every page?

Something like
```
if successPage.startsWith(window.location.pathname + '#') -> scrollTo successPage-pathname
```

Reply:
> I am not too sure to understand what is trying to be achieved here (even before your diff). Is the goal to detect that the success URL is the homepage and that we are on the homepage, and in this case don't perform a redirect but a scrollTo (note that the browser don't actually reload but do an ugly scrollto himself)?
> 
> If that's the case, why is this code not doing that for every page but only the homepage? thinking Very louch to do that only for homepage while it could work on every page?
> 
> Something like
> 
> ```
> if successPage.startsWith(window.location.pathname + '#') -> scrollTo successPage-pathname
> ```

Actually when the anchor link option creates the anchor it appends the pathname to the window location before the hash so I thought it's not obvious for the user that if we are on the home page, pasting something like "/#anchor-name" is not an anchor but a redirect to the same page. While something like "/contactus#anchor-name", will make it clear that the link is a page.

But I agree that we can do it for all pages, it makes more sense.